enigma/src/main/java/cuchaz/enigma/EnigmaProject.java

```diff
@@ -344,6 +344,11 @@ public static class ClassSource {
 		}
 
 		public void writeTo(Path path) throws IOException {
+			if(!Files.exists(path)) {
```

Comment:
The Checkstyle step on the CI build workflow will fail due to a missing space in the `if` statement you added. It'd be nice to fix that.

```suggestion
			if (!Files.exists(path)) {
```